sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/SystemPropertyAndEnvironmentLookup.java

```diff
@@ -1,19 +1,78 @@
 package org.sdase.commons.server.dropwizard.bundles;
 
-import io.dropwizard.configuration.EnvironmentVariableLookup;
+import com.fasterxml.jackson.core.io.CharTypes;
+import java.util.Optional;
+import org.apache.commons.text.lookup.StringLookup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Lookup for Java's system properties and environment variables. System properties have higher
  * priority.
  */
-public class SystemPropertyAndEnvironmentLookup extends EnvironmentVariableLookup {
+public class SystemPropertyAndEnvironmentLookup implements StringLookup {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(SystemPropertyAndEnvironmentLookup.class);
+
+  private static final String TO_JSON_STRING_OPERATOR_NORMALIZED = "toJsonString".toLowerCase();
+
+  /**
+   * Provides the actual key to lookup without any operators.
+   *
+   * @param definedKey The key with operators as defined in the String with substitutions. Examples:
+   *     <ul>
+   *       <li><code>foo: ${FOO_PROPERTY}</code> => definedKey = {@code FOO_PROPERTY}
+   *       <li><code>foo: ${FOO_PROPERTY:-default}</code> => definedKey = {@code FOO_PROPERTY}
+   *       <li><code>foo: ${FOO_PROPERTY | toJsonString:-default}</code> => definedKey = {@code
+   *           FOO_PROPERTY | toJsonString}
+   *     </ul>
+   *
+   * @return The key that should be looked up in the environment. All example for {@code definedKey}
+   *     above will return {@code FOO_PROPERTY}.
+   */
+  public static String extractKeyToLookup(String definedKey) {
+    return splitKeyInKeyAndOperators(definedKey)[0].trim();
+  }
```

Comment:
This could be helpful for extracting all keys that are defined in a config file, e.g. for documentation purposes.